tfhe/src/core_crypto/algorithms/lwe_encryption.rs

```diff
@@ -417,21 +535,58 @@ where
 
     let (mask, body) = lwe_ciphertext.get_mask_and_body();
 
-    if ciphertext_modulus.is_native_modulus() {
-        Plaintext((*body.data).wrapping_sub(slice_wrapping_dot_product(
+    match ciphertext_modulus.kind() {
+        CiphertextModulusKind::Native => Plaintext((*body.data).wrapping_sub(
```

Comment:
Could factor ` Plaintext((*body.data).wrapping_sub(` out of the `match`

Reply:
I don't think it helps readability having a Plaintext around the match statement

Reply:
Me neither, but it could be stored in a variable before being wrapped in a `Plaintext`

Reply:
then rust will complain of an unnecessary binding or something, so disagree on that one

Reply:
Something like
```
let dot_product = match {};
Plaintext((*body.data).wrapping_sub(dot_product));
```
I don´t see why Rust would complain

Reply:
I think this is ok and factorizes what can be

```rust
    let mask_key_dot_product = slice_wrapping_dot_product(mask.as_ref(), lwe_secret_key.as_ref());
    let plaintext = (*body.data).wrapping_sub(mask_key_dot_product);

    match ciphertext_modulus.kind() {
        CiphertextModulusKind::Native => Plaintext(plaintext),
        CiphertextModulusKind::NonNativePowerOfTwo => {
            // Manage power of 2 encoding
            Plaintext(
                plaintext
                    .wrapping_div(ciphertext_modulus.get_power_of_two_scaling_to_native_torus()),
            )
        }
        CiphertextModulusKind::Other => unreachable!(),
    }
```

Reply:
LGTM